packages/flutter_ai_providers/lib/src/providers/openwebui.dart

```diff
@@ -0,0 +1,307 @@
+  }): _history = history?.toList() ?? [],
+      _model = model,
+      _host = baseUrl,
+      _apiKey = apiKey;
+
+  final List<ChatMessage> _history;
+  final String _model;
+  final String _host;
+  final String? _apiKey;
+  final List<_OwuiFileAttachment> _fileAttachments = [];
+  final List<_OwuiImageAttachment> _imageAttachments = [];
+  final _emptyMessage = ChatMessage(origin: MessageOrigin.llm, text: null, attachments: []);
+
+  @override
+  Stream<String> generateStream(
+    String prompt, {
+    Iterable<Attachment> attachments = const [],
+  }) async* {
+    final userMessage = ChatMessage(text: prompt, attachments: attachments, origin: MessageOrigin.user);
+    final llmMessage = ChatMessage(text: "", attachments: [], origin: MessageOrigin.llm);
+    
+    yield* _generateStream([userMessage, llmMessage]);
+    
+    notifyListeners();
+  }
+
+  @override
+  Stream<String> sendMessageStream(
+    String prompt, {
+    Iterable<Attachment> attachments = const [],
+  }) async* {
+    final userMessage = ChatMessage(text: prompt, attachments: attachments, origin: MessageOrigin.user);
+    final llmMessage = ChatMessage(text: null, attachments: [], origin: MessageOrigin.llm);
+    _history.addAll([userMessage, llmMessage]);
+
+    yield* _generateStream(_history);
+    notifyListeners();
+  }
+
+  Stream<String> _generateStream(List<ChatMessage> messages) async* {
+    _imageAttachments.clear();
+    _fileAttachments.clear();
+    
+    final files = messages.lastWhere((m) => m.origin == MessageOrigin.user, orElse: () => _emptyMessage).attachments;
+    final llmMessage = messages.last;
+    if(files.isNotEmpty) {
+      for (var file in files) {
+        await _handleAttachment(file);
+      }
+    }
+
+    final httpRequest = http.Request('POST', Uri.parse("$_host/chat/completions"))
+      ..headers.addAll({
+        if(_apiKey != null) 'Authorization': 'Bearer $_apiKey',
+        'Content-Type': 'application/json',
+      })
+      ..body = _OwuiChatRequest(
+        model: _model,
+        messages: messages.where((m) => m.text != null).toList(),
+        files: _fileAttachments,
+        images: _imageAttachments,
+      ).toJsonString();
+
+    final httpResponse = await http.Client().send(httpRequest);
+
+    if (httpResponse.statusCode == 200) {
+      await for (var text in httpResponse.stream.transform(utf8.decoder)) {
+        final messages = text.split('\n');
+        for (var message in messages) {
+          if (message.startsWith('data: [DONE]')) {
+            return;
+          }
+          if (message.isEmpty) continue;
+          final cleanedMessage = message.replaceFirst('data: ', '').trim();
+          try {
+            final chatResponse = _OwuiChatResponse.fromJsonString(cleanedMessage);
+            for (var choice in chatResponse.choices) {
+              llmMessage.append(choice.message.text ?? '');
+              yield choice.message.text ?? '';
+            }
+          } catch (e) {
```

Comment:
I think we can propagate the error instead of hiding it. That's what the rest of providers do.

Reply:
Done